components/drivers/include/drivers/pin.h

```diff
@@ -49,41 +49,41 @@ struct rt_device_pin
 
 struct rt_device_pin_mode
 {
-    rt_uint16_t pin;
-    rt_uint16_t mode;
+    rt_base_t pin;
+    rt_uint8_t mode;
 };
 struct rt_device_pin_status
 {
-    rt_uint16_t pin;
+    rt_base_t pin;
     rt_uint16_t status;
 };
 struct rt_pin_irq_hdr
 {
-    rt_int16_t        pin;
-    rt_uint16_t       mode;
+    rt_base_t        pin;
+    rt_uint8_t       mode;
     void (*hdr)(void *args);
     void             *args;
 };
 struct rt_pin_ops
 {
-    void (*pin_mode)(struct rt_device *device, rt_base_t pin, rt_base_t mode);
-    void (*pin_write)(struct rt_device *device, rt_base_t pin, rt_base_t value);
+    void (*pin_mode)(struct rt_device *device, rt_base_t pin, rt_uint8_t mode);
+    void (*pin_write)(struct rt_device *device, rt_base_t pin, rt_uint8_t value);
     int (*pin_read)(struct rt_device *device, rt_base_t pin);
-    rt_err_t (*pin_attach_irq)(struct rt_device *device, rt_int32_t pin,
-                      rt_uint32_t mode, void (*hdr)(void *args), void *args);
-    rt_err_t (*pin_detach_irq)(struct rt_device *device, rt_int32_t pin);
+    rt_err_t (*pin_attach_irq)(struct rt_device *device, rt_base_t pin,
+            rt_uint8_t mode, void (*hdr)(void *args), void *args);
+    rt_err_t (*pin_detach_irq)(struct rt_device *device, rt_base_t pin);
     rt_err_t (*pin_irq_enable)(struct rt_device *device, rt_base_t pin, rt_uint32_t enabled);
     rt_base_t (*pin_get)(const char *name);
 };
 
 int rt_device_pin_register(const char *name, const struct rt_pin_ops *ops, void *user_data);
 
-void rt_pin_mode(rt_base_t pin, rt_base_t mode);
-void rt_pin_write(rt_base_t pin, rt_base_t value);
+void rt_pin_mode(rt_base_t pin, rt_uint8_t mode);
+void rt_pin_write(rt_base_t pin, rt_uint8_t value);
 int  rt_pin_read(rt_base_t pin);
-rt_err_t rt_pin_attach_irq(rt_int32_t pin, rt_uint32_t mode,
+rt_err_t rt_pin_attach_irq(rt_base_t pin, rt_uint8_t mode,
                              void (*hdr)(void *args), void  *args);
-rt_err_t rt_pin_detach_irq(rt_int32_t pin);
+rt_err_t rt_pin_detach_irq(rt_base_t pin);
 rt_err_t rt_pin_irq_enable(rt_base_t pin, rt_uint32_t enabled);
 /* Get pin number by name,such as PA.0,P0.12 */
 rt_base_t rt_pin_get(const char *name);
```

Comment:
/* Get pin number by name, such as PA.0, P0.12 */ 这个注释放到.c这个函数的脑袋顶上就可以了

```suggestion
rt_base_t rt_pin_get(const char *name);
```